core/test/financial-templates/Liquidatable.js

```diff
@@ -1215,4 +1216,335 @@ contract("Liquidatable", function(accounts) {
+        const sponsorUSDCBalanceBefore = await collateralToken.balanceOf(sponsor);
+        await USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: sponsor });
+        const sponsorUSDCBalanceAfter = await collateralToken.balanceOf(sponsor);
+
+        // Expected Sponsor payment => remaining collateral (locked collateral - TRV) + sponsor reward
+        const USDCExpectedPayment = USDCAmountOfCollateral.sub(USDCSettlementTRV).add(USDCSponsorDisputeReward);
+        assert.equal(sponsorUSDCBalanceAfter.sub(sponsorUSDCBalanceBefore).toString(), USDCExpectedPayment.toString());
+
+        // Sponsor should not be able to call again
+        assert(
+          await didContractThrow(
+            USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: sponsor })
+          )
+        );
+      });
+      it("Liquidator calls", async () => {
+        const liquidatorUSDCBalanceBefore = await collateralToken.balanceOf(liquidator);
+        await USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, {
+          from: liquidator
+        });
+        const liquidatorUSDCBalanceAfter = await collateralToken.balanceOf(liquidator);
+
+        // Expected Liquidator payment => TRV - dispute reward - sponsor reward
+        const expectedPayment = USDCSettlementTRV.sub(USDTDisputerDisputeReward).sub(USDCSponsorDisputeReward);
+        assert.equal(
+          liquidatorUSDCBalanceAfter.sub(liquidatorUSDCBalanceBefore).toString(),
+          expectedPayment.toString()
+        );
+
+        // Liquidator should not be able to call again
+        assert(
+          await didContractThrow(
+            USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: liquidator })
+          )
+        );
+      });
+      it("Disputer calls", async () => {
+        const disputerUSDCBalanceBefore = await collateralToken.balanceOf(disputer);
+        await USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: disputer });
+        const disputerUSDCBalanceAfter = await collateralToken.balanceOf(disputer);
+
+        // Expected Disputer payment => disputer reward + dispute bond
+        const expectedPayment = USDTDisputerDisputeReward.add(USDCDisputeBond);
+        assert.equal(disputerUSDCBalanceAfter.sub(disputerUSDCBalanceBefore).toString(), expectedPayment.toString());
+
+        // Disputer should not be able to call again
+        assert(
+          await didContractThrow(
+            USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: disputer })
+          )
+        );
+      });
+      it("Rando calls", async () => {
+        assert(
+          await didContractThrow(
+            USDCLiquidationContract.withdrawLiquidation(liquidationParams.liquidationId, sponsor, { from: rando })
+          )
+        );
+      });
```

Comment:
nit: are these tests relevant to the decimalization?

Reply:
it's not directly relevant but I wanted to execute a full lifecycle test to ensure that all core logic still operated exactly as expected. If I was to focus on the decimalization only then I'd remove any test that does not involve an assertion on a numerical calculation.